qualtran/bloqs/arithmetic/addition.py

```diff
@@ -74,18 +75,26 @@ class Add(Bloq):
         [Halving the cost of quantum addition](https://arxiv.org/abs/1709.06648)
     """
 
-    dtype: Union[QInt, QUInt, QMontgomeryUInt] = field()
+    a_dtype: Union[QInt, QUInt, QMontgomeryUInt] = field()
+    b_dtype: Union[QInt, QUInt, QMontgomeryUInt] = field()
 
-    @dtype.validator
-    def _dtype_validate(self, field, val):
+    @a_dtype.validator
+    def _a_dtype_validate(self, field, val):
+        if not isinstance(val, (QInt, QUInt, QMontgomeryUInt)):
+            raise ValueError("Only QInt, QUInt and QMontgomerUInt types are supported.")
+        if isinstance(val.num_qubits, sympy.Expr):
```

Comment:
>         if isinstance(val.num_qubits, sympy.Expr):
>              return
this check does nothing

Reply:
Ya this was there before so I just didn't remove it because I wasn't sure but I'll remove it now.